torch_xla/csrc/tensor.cpp

```diff
@@ -632,22 +635,29 @@ std::string XLATensor::DumpHloComputation(
 }
 
 XLATensor::ShardingSpecPtr XLATensor::sharding_spec() const {
-  XLA_CHECK(GetIrValue().node != nullptr) << "Tyring to access a null cursor";
-  auto* sharding =
-      dynamic_cast<XlaNode*>(data()->ir_value.node.get())->GetSharding();
-  if (sharding == nullptr) {
-    return nullptr;
+  torch::lazy::Value ir_value = CurrentIrValue();
+  if (ir_value) {
+    XLA_CHECK(ir_value.node != nullptr) << "Tyring to access a null cursor";
+    auto* sharding = dynamic_cast<XlaNode*>(ir_value.node.get())->GetSharding();
+    if (sharding == nullptr) {
+      return nullptr;
+    }
+    return std::make_shared<ShardingSpec>(*sharding);
   }
-  return std::make_shared<ShardingSpec>(*sharding);
+  return nullptr;
 }
+
 void XLATensor::SetShardingSpec(const ShardingSpec& sharding_spec) {
   XLA_CHECK(GetIrValue().node != nullptr) << "Tyring to access a null cursor";
-  dynamic_cast<XlaNode*>(data()->ir_value.node.get())
+  dynamic_cast<XlaNode*>(GetIrValue().node.get())
```

Comment:
hmm, we should add a `XlaNodeCast` to replace `dynamic_cast<XlaNode*>` so it is cleaner

Reply:
I see, I normally prefer more explicit type identifiers especially for casting (similar to avoid using auto too much).